src/components/Map/Map.tsx

```diff
@@ -53,9 +53,9 @@ export default function Map(props: MapProps): JSX.Element {
   const [deferredFocusEntity, setDeferredFocusEntity] = useState<MapEntityId[] | undefined>();
   const mapRef = useRef(null);
   const containerRef = useRef(null);
-  const hoverStateId: { [key: string]: number | undefined } = useMemo(() => ({}), []);
-  const selectStateId: { [key: string]: number | undefined } = useMemo(() => ({}), []);
-  const highlightStateId: { [key: string]: number | undefined } = useMemo(() => ({}), []);
+  const hoverStateId: { [key: string]: { [key: string]: number | undefined } } = useMemo(() => ({}), []);
+  const selectStateId: { [key: string]: { [key: string]: number | undefined } } = useMemo(() => ({}), []);
+  const highlightStateId: { [key: string]: { [key: string]: number | undefined } } = useMemo(() => ({}), []);
```

Comment:
Extract this out as a type? I didn't know about `Record` when I first wrote it.. and was also probably lazy to extract it out earlier.
perhaps
```
  type SomethingState = Record<string, Record<string, numer | undefined>>;
```

Reply:
That's a good call 👍